backend/src/packages/chaiNNer_standard/image_utility/compositing/blend_images.py

```diff
@@ -44,53 +196,88 @@ def blend_images_node(
     base: np.ndarray | Color,
     ov: np.ndarray | Color,
     blend_mode: BlendMode,
+    overlay_position: BlendOverlayPosition,
+    x_percent: int,
+    y_percent: int,
+    x_px: int,
+    y_px: int,
+    crop_to_fit: bool,
 ) -> np.ndarray:
     """Blend images together"""
 
-    # convert colors to images
+    # Convert colors to images
+    do_crop_to_fit = crop_to_fit
     if isinstance(base, Color):
         if isinstance(ov, Color):
             raise ValueError("At least one layer must be an image")
         base = base.to_image(width=ov.shape[1], height=ov.shape[0])
+        do_crop_to_fit = False
     if isinstance(ov, Color):
         ov = ov.to_image(width=base.shape[1], height=base.shape[0])
+        do_crop_to_fit = False
 
-    b_h, b_w, b_c = get_h_w_c(base)
-    o_h, o_w, _ = get_h_w_c(ov)
-    max_h = max(b_h, o_h)
-    max_w = max(b_w, o_w)
+    base_height, base_width, base_channel_count = get_h_w_c(base)
+    overlay_height, overlay_width, _ = get_h_w_c(ov)
 
-    if (b_w, b_h) == (o_w, o_h):
-        # we don't have to do any size adjustments
-        return blend_images(ov, base, blend_mode)
+    # Calculate coordinates of the overlay layer
+    #   origin: top-let point of the base layer
+    #   (x0, y0): top-left point of the overlay layer
+    #   (x1, y1): bottom-right point of the overlay layer
+    if overlay_position == BlendOverlayPosition.PERCENT_OFFSET:
+        v = [
+            round((base_width - overlay_width) * x_percent / 100),
+            round((base_height - overlay_height) * y_percent / 100),
+        ]
+    else:
+        v = [base_width - overlay_width, base_height - overlay_height]
+    x0, y0 = np.array(
+        (
+            BLEND_OVERLAY_X0_Y0_POSITION[overlay_position][0] * v
+            + BLEND_OVERLAY_X0_Y0_POSITION[overlay_position][1] * [x_px, y_px]
```

Comment:
You cannot use `x_px` and `y_px` here. Unless `overlay_position == BlendOverlayPosition.PIXEL_OFFSET` their value is not defined and may be any value allowed by the input. 

This is because we keep the input values of hidden inputs. E.g. if the user selects "Offset (pixels)" and sets the x offset to 200, but then changes their mind and selects "Offset (%)" instead, then `x_px` will still have the value 200.

Reply:
Unless the x_px/y_px are None, this is working because `BLEND_OVERLAY_X0_Y0_POSITION[overlay_position][1]` is equal to `[0,0]` except when overlay_position==BlendOverlayPosition.PIXEL_OFFSET.
I wanted to have one single formula to avoid too many if...elif...else. PERCENT_OFFSET is unfortunately an exception.

Because it is unclear for quick understanding, I may change this to 
```python
    if overlay_position == BlendOverlayPosition.PERCENT_OFFSET:
        x0, y0 = [
            round((base_width - overlay_width) * x_percent / 100),
            round((base_height - overlay_height) * y_percent / 100),
        ]
    elif overlay_position == BlendOverlayPosition.PIXEL_OFFSET:
        x0, y0 = [x_px, y_px]        
    else:
        v = [base_width - overlay_width, base_height - overlay_height]
        x0, y0 = np.array(BLEND_OVERLAY_X0_Y0_POSITION[overlay_position][0] * v).astype("int")
```
What is your opinion?

Reply:
> I may change this to

That is a lot more understandable. Let's go with that.